cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -59,81 +60,51 @@ func isKubeadmConfigPresent(docmap kubeadmapi.DocumentMap) bool {
-		cfg, err := configutil.BytesToInitConfiguration(configBytes, false)
-		return cfg, true, err
+	// `dryRun` should be always `false` for `kubeadm plan`.
+	isDryRun := ptr.To(false)
+	printConfigCfg := upgradeCfg.Plan.PrintConfig
+	ignoreErrCfg := upgradeCfg.Plan.IgnorePreflightErrors
+	if upgradeApply {
+		printConfigCfg = upgradeCfg.Apply.PrintConfig
+		ignoreErrCfg = upgradeCfg.Apply.IgnorePreflightErrors
+		isDryRun = cmdutil.ValueFromFlagsOrConfig(flagSet, options.DryRun, upgradeCfg.Apply.DryRun, &dryRun).(*bool)
 	}
 
-	// If no kubeadm config types are present, we assume that there are manually upgraded component configs in the file.
-	// Hence, we load the kubeadm types from the cluster.
-	initCfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, logPrefix, false, true)
+	client, err := getClient(flags.kubeConfigPath, *isDryRun)
 	if err != nil {
-		return nil, false, err
+		return nil, nil, nil, nil, errors.Wrapf(err, "couldn't create a Kubernetes client from file %q", flags.kubeConfigPath)
 	}
 
-	// Stop here if the caller does not want us to load the component configs
-	if !skipComponentConfigs {
-		// Load the component configs with upgrades
-		if err := componentconfigs.FetchFromClusterWithLocalOverwrites(&initCfg.ClusterConfiguration, client, docmap); err != nil {
-			return nil, false, err
-		}
-
-		// Now default and validate the configs
-		componentconfigs.Default(&initCfg.ClusterConfiguration, &initCfg.LocalAPIEndpoint, &initCfg.NodeRegistration)
-		if errs := componentconfigs.Validate(&initCfg.ClusterConfiguration); len(errs) != 0 {
-			return nil, false, errs.ToAggregate()
-		}
+	ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(flags.ignorePreflightErrors, ignoreErrCfg)
+	if err != nil {
+		return nil, nil, nil, nil, err
 	}
 
-	return initCfg, false, nil
-}
-
-type LoadConfigFunc func(cfgPath string, client clientset.Interface, skipComponentConfigs bool, printer output.Printer) (*kubeadmapi.InitConfiguration, bool, error)
-
-// enforceRequirements verifies that it's okay to upgrade and then returns the variables needed for the rest of the procedure
-func enforceRequirements(flags *applyPlanFlags, args []string, dryRun bool, upgradeApply bool, printer output.Printer, loadConfig LoadConfigFunc) (clientset.Interface, upgrade.VersionGetter, *kubeadmapi.InitConfiguration, error) {
-	client, err := getClient(flags.kubeConfigPath, dryRun)
-	if err != nil {
-		return nil, nil, nil, errors.Wrapf(err, "couldn't create a Kubernetes client from file %q", flags.kubeConfigPath)
+	// Also set the union of pre-flight errors to UpgradeConfiguration, to provide a consistent view of the runtime configuration.
+	// .Plan.IgnorePreflightErrors is not set as it's not used.
+	if upgradeApply {
+		upgradeCfg.Apply.IgnorePreflightErrors = sets.List(ignorePreflightErrorsSet)
```

Comment:
is this causing a problem otherwise?
can we just check if `len(upgradeCfg.Apply.IgnorePreflightErrors) > 0` instead of defaulting it with a `sets.List()` here?

Reply:
I am not sure understanding it correctly, the line here is assigning value to (convert sets.Set[string] to []string) to `upgradeCfg.Apply.IgnorePreflightErrors`, not sure what the check you meant here, correct me pls.